src/priceClient/adapters/baseAdapter.ts

```diff
@@ -13,8 +17,8 @@ export class BaseHTTPAdapter {
     this._timeout = timeout;
   }
 
-  constructor(public readonly name: string, public readonly host: string, { timeout }: { timeout?: number }) {
-    this._timeout = timeout ?? 1000; // ms
+  constructor(public readonly name: string, public readonly host: string, { timeout = 1000 }: BaseHTTPAdapterArgs) {
+    this.timeout = timeout; // ms
```

Comment:
Should this be `this._timeout`?

Reply:
> I assume it's supposed to go through the setter that has the automatic check?

Yep, correct. It was actually an error that it didn't do this beforehand, because it meant that the initial values passed to the constructor were never validated.